ui_test/src/lib.rs

```diff
@@ -426,7 +409,28 @@ fn check_annotations(
             lowest_annotation_level = std::cmp::min(lowest_annotation_level, level);
         }
 
-        if let Some(msgs) = messages.get_mut(line) {
+        let (line, definition_line) = match lines {
+            &ErrorLines::Single(ErrorLine { line, definition_line }) => {
+                let line = messages.get_mut(line).map(|msgs| (msgs, line));
+                (line, definition_line)
+            },
+            ErrorLines::Multiple { lines, .. } => {
+                let mut iter = lines.iter();
+                // Need to loop manually, because borrow checker thinks we borrow `messages` mutably twice with for loops or iterators.
+                let found = loop {
+                    let line = match iter.next() {
+                        None => break None,
+                        Some(line) => line.line,
+                    };
```

Comment:
```suggestion
                    let Some(line) = iter.next() else {
                        break None
                    };
```

Reply:
The ui_test crate is entirely stable, we can't use this yet